wyzesense2mqtt/wyzesense2mqtt.py

```diff
@@ -424,52 +521,118 @@ def on_message_reload(MQTT_CLIENT, userdata, msg):
     init_sensors()
 
 
+# Process message to set keypad state
+def on_message_set(MQTT_CLIENT, userdata, msg):
+    LOGGER.info(f"In on_message_set: {msg.payload.decode()}")
+    parts = msg.topic.split('/')
+    payload = json.loads(msg.payload)
+    if payload and payload['mode'] is not None:
+        set_keypad_mode(parts[1], payload)
+
+
+def set_keypad_mode(keypad_mac, payload):
+    mode = payload['mode']
+    pin = payload['pin']
+
+    if not validate_pin_entry(keypad_mac, mode, pin):
+        return
+
+    mode_payload = {'state': mode}
+
+    mode_topic = f"{CONFIG['self_topic_root']}/{keypad_mac}/mode"
+    set_topic = f"{CONFIG['self_topic_root']}/{keypad_mac}/set"
+
+    mqtt_publish(mode_topic, mode_payload)
+    mqtt_publish(set_topic, {'mode': None, 'pin': False})
+
+
+def validate_pin_entry(mac, mode, pin=None):
+    pin_topic = f"{CONFIG['self_topic_root']}/{mac}/pin"
+    config = SENSORS[mac]
+
+    if pin is None:
+        pin = mqtt_simple_subscribe(pin_topic)['state']
+
+    if pin != config['pin']:
```

Comment:
This makes sure that the PIN entered either matches the one from the `sensors.yaml` file, or if it doesn't, that the mode trying to be activated doesn't require a PIN.